src/native/libs/System.IO.Compression.Native/CMakeLists.txt

```diff
@@ -36,6 +36,12 @@ set(NATIVECOMPRESSION_SOURCES
     pal_zlib.c
 )
 
+if (HOST_WIN32 OR CLR_CMAKE_TARGET_WIN32)
+    list(APPEND NATIVECOMPRESSION_SOURCES "zlib_allocator_win.c")
+else()
+    list(APPEND NATIVECOMPRESSION_SOURCES "zlib_allocator_unix.c")
```

Comment:
We should not be adding the custom allocator back on unix unless we have evidence that it improves performance.

I see this change as a workaround for a poor implementation of standard allocator on Windows.

Reply:
I can test Linux with the same repo benchmark that showed the regression on Windows to see if it's worth it to bring back there. 

It sounds like you're making a case to reduce the allocator (removing canary check).  I can do this - simplifying the custom implementation down to just a dedicated heap.  Initially I had not done this because I was opting for the simple revert to a known shipping state.  Sounds like you'd prefer a new implementation that does less.  It has some risk with it but perhaps we can make it small enough to be happy with such risk in servicing.

Reply:
> simplifying the custom implementation down to just a dedicated heap

Yes, I think that would be best.

Reply:
> low-fragmentation heap

Low fragmentation heap kicks in only for sizes up to 16kB. My guess is that the typical zlib-ng allocation is a lot more than that.

Reply:
Enabling segment heap may help to mitigate the issue: https://learn.microsoft.com/en-us/windows/win32/sbscs/application-manifests#heaptype . Segment heap uses different strategy for allocations >16kB.